### PR TITLE
Add stripping of trailing whitespace to pre-commit

### PR DESCRIPTION
**Pull request by suzil**
### Description

Add a pre-commit hook to strip trailing whitespace. It's commit noise, and this pre-commit hook will keep code consistent by stripping it always.

### CODE REVIEW

**Comment by suzil** [comment]
I'm intrigued as to why CI fails from stripping whitespace 😁

**Comment by erwanp** [comment]
The `.spec` files are how we store our Spectrum objects with metadata. They're either renamed .json, or compressed as binary. Looks like pre-commit [modified the binaries](https://github.com/radis/radis/pull/142/files?file-filters%5B%5D=.spec).

**Comment by suzil** [comment]
Excluded those files:

```yaml
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.3.0
    hooks:
      - id: trailing-whitespace
        exclude: |
            (?x)(
                ^.+.spec|
                ^.+.dat
            )
```

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,5 +8,14 @@ repos:
...
+      - id: flake8
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.3.0
+    hooks:
+      - id: trailing-whitespace
+        exclude: |
+            (?x)(
+                ^.+.spec|
+                ^.+.dat
```
Comment (erwanp) [inline .pre-commit-config.yaml:20]: Hello ! I also excluded the .levels files, which are some other plain text input files. 

```suggestion
                ^.+.dat|
                ^.+.levels
```

**Review by erwanp — CHANGES_REQUESTED** [review]
suggested another change, otherwise we're good to go !

**Comment by suzil** [comment]
```yaml
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.3.0
    hooks:
      - id: trailing-whitespace
        exclude: |
            (?x)(
                ^.+\.spec|
                ^.+\.dat|
                ^.+\.levels
            )
```
- Added `*.levels`
- Fixed regex to escape the period before the file extension

**Pull request merged**